### PR TITLE
feat(lane_change): add param for a lateral distance margin where the abort can be performed

### PR DESCRIPTION
**Pull request by TakaHoribe**
## Description

Add a parameter for the "abort" execution. If the ego is over the lane with the given threshold, abort will not be performed.

launcher: https://github.com/autowarefoundation/autoware_launch/pull/421

## Related links

None

## Tests performed

Run Psim.

In the video below,  the debug marker is temporally visualized. The "abort" can be performed when the polygon is green, and prohibited when it is red.

With margin = 0.0

https://github.com/autowarefoundation/autoware.universe/assets/21360593/96dcb5c9-6ada-4681-966a-a05ff08d8587

With margin = 1.0

https://github.com/autowarefoundation/autoware.universe/assets/21360593/2200b55f-7db2-4dc1-b81c-5474825c7183

Note: the computing time changes slightly. `0.019[ms]` -> `0.030[ms]`

## Notes for reviewers

None

## Interface changes

None

## Effects on system behavior

None

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [x] The PR follows the [pull request guidelines].
- [x] The PR has been properly tested.
- [x] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [x] There are no open discussions or they are tracked via tickets.
- [x] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: planning/behavior_path_planner/config/lane_change/lane_change.param.yaml [hunk 1]
```diff
@@ -53,6 +53,7 @@
       abort_delta_time: 1.0                      # [s]
       aborting_time: 5.0                         # [s]
       abort_max_lateral_jerk: 1000.0             # [m/s3]
+      abort_exec_lateral_threshold: 0.0
```
Comment (TakaHoribe) [inline planning/behavior_path_planner/config/lane_change/lane_change.param.yaml:56]: @zulfaqar-azmi-t4 @rej55 Any idea for a good naming?

Comment (rej55) [reply planning/behavior_path_planner/config/lane_change/lane_change.param.yaml:56]: How is `overhang_threshold_for_continue_lane_change` ?

Comment (rej55) [reply planning/behavior_path_planner/config/lane_change/lane_change.param.yaml:56]: or, `overhang_margin_for_continue_lane_change`

Comment (TakaHoribe) [reply planning/behavior_path_planner/config/lane_change/lane_change.param.yaml]: changed to `lane_change.cancel.overhang_tolerance`

**Pull request merged**